instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/ConfigPropertiesUtil.java

```diff
@@ -17,11 +29,70 @@
  */
 public final class ConfigPropertiesUtil {
 
+  private static final boolean supportsDeclarativeConfig = supportsDeclarativeConfig();
+
+  private static final Map<String, String> config = load();
```

Comment:
is this new static state important?

Reply:
is the performance implementation important for this PR? if not, let's take it out and you can propose it separately